docs/user-guide/src/project-overview.md

```diff
@@ -0,0 +1,65 @@
+# Project overview
+
+Metal3 consists of multiple sub-projects. The most notable are [Bare Metal
+Operator](https://github.com/metal3-io/baremetal-operator), [Cluster API
+provider Metal3](https://github.com/metal3-io/cluster-api-provider-metal3) and
+the [IP address manager](https://github.com/metal3-io/ip-address-manager). There
+is no requirement to use all of them.
+
+The stack, when including Cluster API and Ironic, looks like this:
+
+![Metal3 stack](./images/metal3-stack.png)
+
+From a user perspective it may be more useful to visualize the Kubernetes
+resources. When using Cluster API, Metal3 works as any other infrastructure
+provider. The Machines get corresponding Metal3Machines, which in turn reference
+the BareMetalHosts.
+
+![CAPI-machines](./images/capi-machines.png)
+
+The following diagram shows more details about the Metal3 objects. Note that it
+is not showing everything and is meant just as an overview.
+
+![Metal3-CAPI objects](./images/metal3-capi-objects.png)
+
+## How does it work?
+
+Metal3 relies on Ironic for interacting with the physical machines. Ironic in
+turn communicates with [Baseboard Management
+Controllers](https://en.wikipedia.org/wiki/Intelligent_Platform_Management_Interface#Baseboard_management_controller)
+(BMCs) to manage the machines. Ironic can communicate with the BMCs using
+protocols such as Redfish, IPMI, or iDRAC. In this way, it can power on or off
+the machines, change the boot device, and so on.
+
+For more advanced operations, like writing an image to the disk, the [Ironic
+Python Agent](https://opendev.org/openstack/ironic-python-agent) (IPA) is first
```

Comment:
nit: link to /ironic/ironic-python-agent instead (the git repo is not too helpful)